### PR TITLE
feat(ext/console): better circular information in object inspection

### PR DESCRIPTION
**Pull request by crowlKats**
example code:
```ts
const x = { a: {}, b: {}, foo: { safdalskdma: "fijnsiuesfniuw", hello: 1 } };
x.a.x = x;
x.b.y = x;
console.log(x);

const y = { a: { b: {} } };
y.a.b = y.a;
console.log(y);
```

Before
Screenshot 2022-02-01 at 02 19 16

After
Screenshot 2022-02-01 at 02 20 17

going to look into adding tests later

### CODE REVIEW

File: ext/console/02_console.js [hunk 1]
```diff
@@ -624,10 +641,9 @@
         }
 
         if (ctxHas(value)) {
-          // Circular string is cyan
-          return cyan("[Circular]");
+          return handleCircular(value);
         }
-        return inspectObject(value, level, inspectOptions);
+        return inspectObject(value, level, inspectOptions, circular);
```
Comment (kt3k) [inline ext/console/02_console.js:646]: The 4th arg doesn't seem being used?

**Comment by kt3k** [comment]
The behavior looks good to me. We need to update test cases `consoleTestStringifyCircular`, etc.

File: ext/console/02_console.js (hunk 1 above)
Comment (crowlKats) [reply ext/console/02_console.js:646]: Whoops, that was remainder from some testing

File: ext/console/02_console.js [hunk 2]
```diff
@@ -884,6 +900,7 @@
   }
 
   function inspectError(value) {
+    // TODO(@crowlKats): use circular system
```
Comment (bartlomieju) [inline ext/console/02_console.js:903]: Can this TODO be addressed before landing?

Comment (crowlKats) [reply ext/console/02_console.js:903]: yes, will do

File: cli/tests/unit/console_test.ts [hunk 3]
```diff
@@ -392,14 +392,14 @@ Deno.test(function consoleTestStringifyFunctionWithProperties() {
   assertEquals(
     stringify({ f }),
     `{
-  f: [Function: f] {
+  f: [Function: f] <ref *1> {
```
Comment (crowlKats) [inline cli/tests/unit/console_test.ts:395]: I am not happy with this. In node the ref is before `[Function: f]` or any other description.
This is also inconsistent behaviour as
```js
class Foo {}
const x = new Foo();
x.y = x;
console.log(x);
```
Will output `<ref *1> Foo { y: [Circular *1] }`

Comment (bartlomieju) [reply cli/tests/unit/console_test.ts:395]: Can't `<ref *1>` be prepended in this case?

Comment (kt3k) [reply cli/tests/unit/console_test.ts:406]: Maybe let's make `inspectRawObject` returning `[inspected, refIndex]` array, and prepend `<ref *1>` part inside of `inspectFunction`? (around here: https://github.com/denoland/deno/blob/26f5c223df26962f19ccc957d0507facc09c55c5/ext/console/02_console.js#L358-L362

File: cli/tests/unit/console_test.ts [hunk 4]
```diff
@@ -350,12 +350,23 @@ Deno.test(function consoleTestStringifyCircular() {
         return Deno.inspect(this);
       },
     }),
-    "[Circular]",
+    "[Circular *1]",
```
Comment (crowlKats) [inline cli/tests/unit/console_test.ts:353]: In this cause, shouldnt it rather be `<ref *1> [Circular *1]`?

**Review by kt3k — APPROVED** [review]
LGTM

**Pull request merged**